### ISSUE TITLE
[FEATURE] Add correctness check capability.

### ISSUE THREAD

**Issue by andyw-lala** [HPE/torch-hammer#23]
## Summary

Add the capability to verify correctness of calculation against expected/golden value(s.)

## Motivation

There is a proven need in modern systems to verify correctness in an ongoing manner.
This is accomplished usually by running known code with known inputs and testing the results against acceptable value(s)/range(s.)

## Proposed Solution

TBD.

## Alternatives Considered

This functionality is currently implemented in node-check regimes at multiple sites, using tools such as HPL and checking the residual against a golden value.

## Would you be willing to contribute this feature?

- [x] Yes, I'd like to implement this
- [x] Yes, with guidance
- [x] I can help test, regardless of who develops

**Comment by hagertnl** [comment]
I would also be willing to contribute to this feature as we investigate using torch-hammer on Frontier in testing.

In other AI-related tests, we typically look for reasonably well-converging/similar loss functions in training workloads. In this case, a `--diff` option for correctness-related values between 2 runs may be sufficient -- over time, if 500 runs all validate to a "reference" run, then it becomes a useful "golden" value as Andy mentioned. We do this for HPL, LAMMPS, AMG, among other tests already.

**Comment by andyw-lala** [comment]
I will work to keep a potential known failing node in a test system as we develop and test this feature, though obviously we cannot tie up a failing node indefinitely.
